idaes/models/costing/SSLW.py

```diff
@@ -1016,13 +1016,24 @@ def cost_compressor(
             integer: whether the number of units should be constrained to be
                 an integer or not (default = True).
         """
-        # Confirm that unit is a turbine
+        # Confirm that unit is a compressor
         if not blk.unit_model.config.compressor:
             raise TypeError(
                 "cost_compressor method is only appropriate for "
                 "pressure changers with the compressor argument "
                 "equal to True."
             )
+        # compressor = True, and using non-isentropic assumption
+        # (costing not needed)
+        elif hasattr(blk.unit_model.config, "thermodynamic_assumption"):
+            if (blk.unit_model.config.thermodynamic_assumption.name) != "isentropic":
+                # PYLINT-TODO-FIX fix exception message with correct number of arguments
+                raise ValueError(
+                    "{} - pressure changers without isentrpoic "  # pylint: disable=E1305
+                    "assumption are too simple to be costed. ".format(
+                        blk.unit_model.name
+                    )
+                )
```

Comment:
```suggestion
            if blk.unit_model.config.thermodynamic_assumption.name != "isentropic":
                raise ValueError(
                    f"{blk.unit_model.name} - pressure changers without isentropic "
                    "assumption are too simple to be costed."
                )
```

Reply:
Suggestions for minor enhancements:

- I don't think the `# pylint:` directives here are needed anymore, so we can get rid of them
- Using f-strings instead of `.format()` is more readable
- Typo: `isentrpoic` in exception message